### PR TITLE
Extend teams stats API with team members stats API

### PR DESCRIPTION
**Pull request by ValentinTrinque**

### CODE REVIEW

File: datanode/gateway/graphql/schema.graphql [hunk 1]
```diff
@@ -1024,6 +1024,22 @@ type Query {
     pagination: Pagination
   ): TeamsStatisticsConnection
 
+  """
+  List team members statistics for a given team
+  Get the statistics of all team members for a given team ID, or for a specific member by using party ID, over a number of epochs.
+  If a team does not have at least the number of epochs worth of data, it is ignored.
+  """
+  teamMembersStatistics(
+    "team ID to filter for"
```
Comment (candida-d) [inline datanode/gateway/graphql/schema.graphql:1033]:
```suggestion
    "Team ID to filter for"
```

File: datanode/gateway/graphql/schema.graphql [hunk 2]
```diff
@@ -1024,6 +1024,22 @@ type Query {
     pagination: Pagination
   ): TeamsStatisticsConnection
 
+  """
+  List team members statistics for a given team
```
Comment (candida-d) [inline datanode/gateway/graphql/schema.graphql:1028]:
```suggestion
  List team members' statistics for a given team
```

File: protos/sources/data-node/api/v2/trading_data.proto [hunk 3]
```diff
@@ -819,6 +819,14 @@ service TradingDataService {
     option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {tags: "Teams"};
   }
 
+  // List team members statistics
```
Comment (candida-d) [inline protos/sources/data-node/api/v2/trading_data.proto:822]:
```suggestion
  // List team members' statistics
```

File: protos/sources/data-node/api/v2/trading_data.proto [hunk 4]
```diff
@@ -819,6 +819,14 @@ service TradingDataService {
...
+  //
+  // Get the statistics of all members of a given team, or for a specific member by using party ID, over a number of epochs.
```
Comment (candida-d) [inline protos/sources/data-node/api/v2/trading_data.proto:824]:
```suggestion
  // Get the statistics for all members of a given team, or for a specific member by using party ID, over a number of epochs.
```

File: protos/sources/data-node/api/v2/trading_data.proto [hunk 5]
```diff
@@ -4188,6 +4196,60 @@ message QuantumRewardsPerEpoch {
...
 }
 
+// Request to list all team members' statistics.
+message ListTeamMembersStatisticsRequest {
+  // Restrict team statistics to those with the given team ID.
+  string team_id = 1;
+  // Restrict team members' statistics to those with the given party ID.
+  optional string party_id = 2;
+  // Defines the number of past epochs to aggregate data from. By default, it takes
+  // the last 10 epochs.
```
Comment (candida-d) [inline protos/sources/data-node/api/v2/trading_data.proto:4206]: This says 30 epochs in the GQL docs. Are they actually different?

File: protos/sources/data-node/api/v2/trading_data.proto [hunk 6]
```diff
@@ -4188,6 +4196,60 @@ message QuantumRewardsPerEpoch {
...
+  // Total of volume accumulated over the requested epoch period, expressed in
+  // quantum value.
+  string total_quantum_volume = 2;
+  // Total of rewards accumulated over the requested epoch period, expressed in
+  // quantum value.
+  string total_quantum_rewards = 3;
+  // List of rewards over the requested epoch period, expressed in quantum
+  // value for each epoch.
+  repeated QuantumRewardsPerEpoch quantum_rewards = 4;
+  // Total of games played.
```
Comment (candida-d) [inline protos/sources/data-node/api/v2/trading_data.proto:4247]:
```suggestion
  // Total number of games played.
```

File: protos/sources/data-node/api/v2/trading_data.proto [hunk 7]
```diff
@@ -4188,6 +4196,60 @@ message QuantumRewardsPerEpoch {
...
+  uint64 total_game_played = 5;
```
Comment (candida-d) [inline protos/sources/data-node/api/v2/trading_data.proto:4248]: wouldn't this be better as "total_games_played"?

File: datanode/gateway/graphql/schema.graphql [hunk 8]
```diff
@@ -6533,6 +6549,38 @@ type TeamsStatisticsConnection {
...
+type TeamMemberStatistics {
+  "Party ID the statistics are related to."
+  partyId: String!
+  "Total of volume accumulated over the requested epoch period, expressed in quantum value."
+  totalQuantumVolume: String!
+  "Total of rewards accumulated over the requested epoch period, expressed in quantum value."
+  totalQuantumRewards: String!
+  "List of rewards over the requested epoch period, expressed in quantum value for each epoch"
+  quantumRewards: [QuantumRewardsPerEpoch!]!
+  "Total of games played."
```
Comment (candida-d) [inline datanode/gateway/graphql/schema.graphql:6562]:
```suggestion
  "Total number of games played."
```

File: datanode/gateway/graphql/schema.graphql [hunk 9]
```diff
@@ -6533,6 +6549,38 @@ type TeamsStatisticsConnection {
...
+  totalGamePlayed: Int!
```
Comment (candida-d) [inline datanode/gateway/graphql/schema.graphql:6563]: Shouldn't this be "totalGamesPlayed"

File: protos/sources/data-node/api/v2/trading_data.proto (hunk 5 above)
Comment (ValentinTrinque) [reply protos/sources/data-node/api/v2/trading_data.proto:4206]: It should be 10. It's changed.

**Review by guoguojin — DISMISSED** [review]

**Review by guoguojin — DISMISSED** [review]

**Review by candida-d — DISMISSED** [review]

**Review by guoguojin — APPROVED** [review]

**Review by candida-d — APPROVED** [review]

**Pull request merged**